components/eamxx/src/physics/mam/eamxx_mam_constituent_fluxes_interface.cpp

```diff
@@ -0,0 +1,292 @@
+
+  // Cloud liquid number mixing ratio [1/kg](Require only for building DS)
+  add_field<Required>("nc", scalar3d_mid, n_unit, grid_name, "tracers");
+
+  // Cloud ice number mixing ratio [1/kg](Require only for building DS)
+  add_field<Required>("ni", scalar3d_mid, n_unit, grid_name, "tracers");
+
+  // Temperature[K] at midpoints
+  add_field<Required>("T_mid", scalar3d_mid, K, grid_name);
+
+  // Vertical pressure velocity [Pa/s] at midpoints (Require only for building
+  // DS)
+  add_field<Required>("omega", scalar3d_mid, Pa / s, grid_name);
+
+  // Total pressure [Pa] at midpoints
+  add_field<Required>("p_mid", scalar3d_mid, Pa, grid_name);
+
+  // Total pressure [Pa] at interfaces
+  add_field<Required>("p_int", scalar3d_int, Pa, grid_name);
+
+  // Layer thickness(pdel) [Pa] at midpoints
+  add_field<Required>("pseudo_density", scalar3d_mid, Pa, grid_name);
+
+  // Planetary boundary layer height [m] (Require only for building DS)
+  add_field<Required>("pbl_height", scalar2d, m, grid_name);
+
+  // cloud fraction [nondimensional] computed by eamxx_cld_fraction_process
+  add_field<Required>("cldfrac_tot", scalar3d_mid, nondim, grid_name);
+
+  static constexpr Units m2(m * m, "m2");
+  static constexpr Units s2(s * s, "s2");
+
+  // Surface geopotential [m2/s2] (Require only for building DS)
+  add_field<Required>("phis", scalar2d, m2 / s2, grid_name);
+
+  // Constituent fluxes at the surface (gasses and aerosols)
+  //[units: kg/m2/s (mass) or #/m2/s (number)]
+  add_field<Required>("constituent_fluxes", scalar2d_pcnct, kg / m2 / s,
+                      grid_name);
+
+  // ---------------------------------------------------------------------
+  // These variables are "Updated" or inputs/outputs for the process
+  // ---------------------------------------------------------------------
+  // NOTE: Cloud borne aerosols are not updated in this process but are included
+  // to create data structures.
+
+  // interstitial and cloudborne aerosol tracers of interest: mass (q) and
+  // number (n) mixing ratios
+  for(int mode = 0; mode < mam_coupling::num_aero_modes(); ++mode) {
+    // interstitial aerosol tracers of interest: number (n) mixing ratios
+    const char *int_nmr_field_name =
+        mam_coupling::int_aero_nmr_field_name(mode);
+    add_field<Updated>(int_nmr_field_name, scalar3d_mid, n_unit, grid_name,
+                       "tracers");
+
+    // cloudborne aerosol tracers of interest: number (n) mixing ratios
+    // NOTE: DO NOT add cld borne aerosols to the "tracer" group as these are
+    // NOT advected
+    const char *cld_nmr_field_name =
+        mam_coupling::cld_aero_nmr_field_name(mode);
+    add_field<Updated>(cld_nmr_field_name, scalar3d_mid, n_unit, grid_name);
+
+    for(int a = 0; a < mam_coupling::num_aero_species(); ++a) {
+      // (interstitial) aerosol tracers of interest: mass (q) mixing ratios
+      const char *int_mmr_field_name =
```

Comment:
This applies to a few places in the file: how about 
```cpp
std::string_view int_mmr_field_name =...
if (not int_mmr_field_name.empty()) {
...
```
Same thing, in practice, but more readable for a C++ dev maybe? strlen is very C, and always makes me a bit wary working with raw pointers.

Reply:
Thanks @bartgol! I have fixed these in the code. I couldn't use `string_view` as the argument type of the `add_field` function is `string` not `string_view`. If I cast `string_view` to `string`, it will make a copy of the string which (to me) looks like the same thing as declaring it as a `string`. Let me know if there is a better way to handle this.